src/kuzzleDataCollection.js

```diff
@@ -79,52 +80,16 @@ function KuzzleDataCollection(kuzzle, collection, index) {
  * usually a couple of seconds.
  * That means that a document that was just been created won’t be returned by this function.
  *
+ * @deprecated
+ * @see KuzzleDataCollection.prototype.search
+ *
  * @param {object} filters - Filters in Elasticsearch Query DSL format
  * @param {object} [options] - Optional parameters
  * @param {responseCallback} cb - Handles the query response
  */
 KuzzleDataCollection.prototype.advancedSearch = function (filters, options, cb) {
-  var
-    query,
-    self = this;
-
-  if (!cb && typeof options === 'function') {
-    cb = options;
-    options = null;
-  }
-
-  self.kuzzle.callbackRequired('KuzzleDataCollection.advancedSearch', cb);
-
-  query = self.kuzzle.addHeaders({body: filters}, this.headers);
-
-  self.kuzzle.query(this.buildQueryArgs('read', 'search'), query, options, function (error, result) {
-    var
-      response,
-      documents = [];
-
-    if (error) {
-      return cb(error);
-    }
-
-    result.result.hits.forEach(function (doc) {
-      var newDocument = new KuzzleDocument(self, doc._id, doc._source);
-
-      newDocument.version = doc._version;
-
-      documents.push(newDocument);
-    });
-
-    response = {
-      total: result.result.total,
-      documents: documents
-    };
-
-    if (result.result.aggregations) {
-      response.aggregations = result.result.aggregations;
-    }
-
-    cb(null, response);
-  });
+  console.warn('Usage of KuzzleDataCollection.advancedSearch is deprecated. Use KuzzleDataCollection.search instead');
```

Comment:
The log would be triggered at each call of `advancedSearch` ? It might make a lot